community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/state/storeview/DynamicIndexStoreView.java

```diff
@@ -71,12 +77,24 @@ public DynamicIndexStoreView( LabelScanStore labelScanStore, LockService locks,
 
     private boolean useAllNodeStoreScan( int[] labelIds )
     {
-        return ArrayUtils.isEmpty( labelIds ) || isEmptyLabelScanStore() || isNumberOfLabeledNodesExceedThreshold( labelIds );
+        try
+        {
+            return ArrayUtils.isEmpty( labelIds ) || isEmptyLabelScanStore() ||
+                    isNumberOfLabeledNodesExceedThreshold( labelIds );
+        }
+        catch ( Exception e )
+        {
+            log.error( "Can not determine number of labeled nodes, falling back to all nodes scan.", e );
```

Comment:
Would be nice if the message mentioned something about not being able to close the opened `AllEntriesLabelScanReader`.